spatial/r3/box.go

```diff
@@ -0,0 +1,69 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+// Box is a 3D bounding box.
+type Box struct {
+	Min, Max Vec
+}
+
+// CenteredBox creates a Box with a given center and size.
+func CenteredBox(center, size Vec) Box {
+	half := Scale(0.5, size)
+	return Box{Min: Sub(center, half), Max: Add(center, half)}
+}
+
+// Size returns the size of the Box.
+func (a Box) Size() Vec {
+	return Sub(a.Max, a.Min)
+}
+
+// Center returns the center of the Box.
+func (a Box) Center() Vec {
+	return Add(a.Min, Scale(0.5, a.Size()))
+}
+
+// Vertices returns a slice of the 8 vertices
+// corresponding to each of the Box's corners.
```

Comment:
do we want to document or specify the order?
or document it isn't specified?

Reply:
Will document that it is not specified.